appendix-D/01_main-chapter-code/previous_chapters.py

```diff
@@ -20,6 +20,7 @@
 
 class GPTDatasetV1(Dataset):
     def __init__(self, txt, tokenizer, max_length, stride):
+        self.tokenizer = tokenizer
```

Comment:
So to clarify, what I meant is that this `self.tokenizer = tokenizer` needs to be removed, otherwise you won't be able to train the model on multiple GPUs.

Reply:
Ah, okay, I see - I got confused by the different implementations in #116 . Maybe it's the best to decline this PR and just push your changes then.

Reply:
Thanks for bringing this to my attention though. I think it wasn't really a problem except for Ch 5-7 where we are training the model. In Ch 2-4 it wouldn't really matter.

Reply:
Yeah, true.

About the `allowed_special={"<|endoftext|>"}` in Ch 2-4, I have found two more cases that have not yet been updated:
- *ch02/01_main-chapter-code/exercise-solutions.ipynb* (to match with *ch02/01_main-chapter-code/ch02.ipynb* code)
- *ch04/01_main-chapter-code/gpt.py* (technically irrelevant as the dataloader won't be used there, at least as far I can see)

Reply:
Thanks, should be addressed everywhere now.